src/test/scala/com/goyeau/kubernetes/client/operation/MinikubeClientProvider.scala

```diff
@@ -20,24 +23,27 @@ trait MinikubeClientProvider[F[_]] extends BeforeAndAfter { this: Suite =>
   }
 
   def resourceName: String
-  private val clean = kubernetesClient
+
+  private val createNamespace = kubernetesClient
+    .use { implicit client =>
+      for {
+        _ <- client.namespaces.deleteTerminated(resourceName.toLowerCase)
+        _ <- NamespacesApiTest.createChecked[F](resourceName.toLowerCase)
+      } yield ()
+    }
+
+  private val deleteNamespace = kubernetesClient
     .use { client =>
       for {
-        namespaceList <- client.namespaces.list
-        namespacesToDelete = namespaceList.items
-          .map(_.metadata.get.name.get)
-          .filter(_.startsWith(resourceName.toLowerCase))
-        _ <- namespacesToDelete.toList.traverse(client.namespaces.deleteTerminated(_))
+        _ <- client.namespaces.delete(resourceName.toLowerCase)
       } yield ()
     }
 
-  before {
-    ConcurrentEffect[F].toIO(clean).unsafeRunSync()
-  }
+  override def beforeAll(): Unit =
```

Comment:
Namespace is now created/deleted once before/after all test cases are run in the suite.